nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java

```diff
@@ -0,0 +1,580 @@
+                    + "for XML files this will be a XPath expression to a single element, and for CSV files this will be a column name (if a header line is present) "
+                    + "or an column index (0-based).")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor OPERATION = new PropertyDescriptor.Builder()
+            .name("aggregator-operation")
+            .displayName("Operation")
+            .description("Which aggregation operation to perform on the specified field for a batch of flow files. Note that some operations can/must take a parameter, "
+                    + "which is specified in the Operation Parameter property.")
+            .required(true)
+            .allowableValues(COUNT, SUM, AVG, MIN, MAX, CONCAT)
+            .defaultValue(COUNT)
+            .expressionLanguageSupported(false)
+            .build();
+
+    public static final PropertyDescriptor OPERATION_PARAM = new PropertyDescriptor.Builder()
+            .name("aggregator-operation-parameter")
+            .displayName("Operation Parameter")
+            .description("An optional parameter given to the aggregation operation. For COUNT, this is an optional integer value that indicates the number to "
+                    + "increment the count by (defaults to 1). For SUM, MAX, MIN, AVERAGE, the value is ignored. For CONCAT, the value is a string inserted "
+                    + "between each field's value in the intermediateAggregate string.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    // Relationships
+    public static final Relationship REL_AGGREGATE = new Relationship.Builder()
+            .name("aggregate")
+            .description("Successfully created FlowFile with an attribute for the aggregated value.")
+            .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("Once an incoming flow file is processed successfully, it will be routed to this relationship")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("If a FlowFile fails processing for any reason (for example, the FlowFile is not in a valid format or the specified "
+                    + "path does not exist), it will be routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+
+    static {
+        System.setProperty("javax.xml.xpath.XPathFactory:" + NamespaceConstant.OBJECT_MODEL_SAXON, "net.sf.saxon.xpath.XPathFactoryImpl");
```

Comment:
Given that NiFi operates in a single JVM instance do we have any other processor that may be doing the same thing yet with different value?